projects/gui/src/boardview/graphicsboard.cpp

```diff
@@ -108,7 +109,14 @@ void GraphicsBoard::paint(QPainter* painter,
 		rect.moveLeft(rLeft);
 		for (int x = 0; x < m_files; x++)
 		{
-			if ((x % 2) == (y % 2))
+			// referenced coordinates
+			int xorig = m_flipped ? m_files - 1 - x : x;
+			int yorig = m_flipped ? m_ranks - 1 - y : y;
```

Comment:
Hello alwey. Here the line of code `int yorig = m_flipped ? m_ranks - 1 -y : y` is in a reversed order as it is not in consistent with the fen string processing code in the board.cpp.

If you test the current code with a fen string that the wall squares are distributed asymmetric both in x-axis and y-axis. you will notice the difference. And the right order of the line of code should be
`int yorig = m_flipped ? y : m_ranks - 1 - y;`

Reply:
I think the implementation is OK now, please see below.